tp1/tp1.go

```diff
@@ -20,21 +22,95 @@ type Carrito struct {
 	Precio int
 }
 
+const (
+	SHOP_NAME_INDEX = 0
+	PROD_ID_INDEX   = 1
+	PRICE_INDEX     = 2
+)
+
+type Article struct {
+	Id    int
+	Price int
+}
+
+func (a Article) ID() int {
+	return a.Id
+}
+
+func (a Article) Precio() int {
+	return a.Price
+}
+
 // CalcularPrecios recibe un arreglo de los IDs de productos y calcula,
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+	var carrito []Carrito
+	var m = make(map[string]int)
+	for _, productId := range ids {
+		for _, productEntry := range p {
+			productEntryId, err := strconv.Atoi(productEntry[PROD_ID_INDEX])
+			if err != nil {
+				// error handling, not a number?
+			}
+			if productEntryId == productId {
```

Comment:
En go por lo general se maneja el "sad path" primero. Eso quiere decir que buscamos la condicion de corte lo antes posible:
```go
if productEntryId != productId {
    continue
}
// codigo
```
Mejora lo que le dicen "line of sight": https://youtu.be/yeetIgNeIkc?t=256